R/relativeimportance.R

```diff
@@ -243,3 +233,217 @@ extractRegressionInfo <- function(formula, data, weights, type, signs,
+#' @importFrom flipU OutcomeName
+#' @importFrom stats terms.formula
+#' @noRd
+computeJaccardCoefficients <- function(formula, data = NULL, weights, variable.names)
+{
+    processed.data <- subsetDataWeightsAndFormula(formula, data, weights)
+    relevant.data <- processed.data$data
+    weights <- processed.data$weights
+    formula <- processed.data$formula
+
+    outcome.name <- OutcomeName(formula, relevant.data)
+    outcome.variable <- relevant.data[[outcome.name]]
+    predictor.names <- attr(terms.formula(formula, data = relevant.data), "term.labels")
+    predictor.variables <- relevant.data[, predictor.names, drop = FALSE]
+    jaccard.coefficients <- vapply(predictor.variables, singleJaccardCoefficient,
+                                   numeric(1), y = outcome.variable, weights = weights)
+    names(jaccard.coefficients) <- variable.names
+    relative.importance <- prop.table(jaccard.coefficients) * 100
+    list(outcome.variable = outcome.variable,
+         coefficients = jaccard.coefficients,
+         relative.importance = relative.importance,
+         predictor.variables = predictor.variables,
+         weights = weights)
+}
+
+#' Compute the single Jaccard coefficient between x and y
+#' @param x The binary input variable x
+#' @param y The binary input variable y
+#' @param centered A logical to determine if the centered coefficient is returned.
+#'   If \code{TRUE}, then the estimate of the mean is subtracted off the jaccard coefficient.
+#' @noRd
+singleJaccardCoefficient <- function(x, y, centered = FALSE, weights = NULL) {
+    if (is.null(weights))
+        j <- sum(y & x, na.rm = TRUE)/sum(x | y, na.rm = TRUE)
+    else
+        j <- sum(weights[y & x], na.rm = TRUE)/sum(weights[x | y], na.rm = TRUE)
+    if (centered)
+        j <- j - singleJaccardExpectation(x, y)
+    j
+}
+
+# Computes the expected value of the Jaccard coefficient between binary variables x and y
+singleJaccardExpectation <- function(x, y)
+{
+    px <- mean(x, na.rm = TRUE)
+    py <- mean(y, na.rm = TRUE)
+    px * py/(px + py - px * py)
+}
+
+#' @param formula The formula used in the Regression model
+#' @param data The data to compute the calculation on
+#' @param weights A numeric vector of weights
+#' @param variable.names Logical vector to determine if variable names or labels should be used.
+#' @importFrom stats terms.formula
+#' @noRd
+computeJaccardCoefficientOutput <- function(formula, data = NULL, weights, variable.names)
+{
+    jaccard.coef.output <- computeJaccardCoefficients(formula, data, weights, variable.names)
+    # Extract the outcome binary variable, the data.frame of predictor binary variables
+    # the computed jaccard coefficients and the relative importance and possible weights
+    y <- jaccard.coef.output$outcome.variable
+    X <- jaccard.coef.output$predictor.variables
+    jaccard.coefs <- jaccard.coef.output$coefficients
+    names(jaccard.coefs) <- variable.names
```

Comment:
already done inside `computeJaccardCoefficients`

Reply:
Removed redundant line in https://github.com/Displayr/flipRegression/pull/22/commits/3451e7f88bae91b96f7a707632a57e746e5ef83a